src/main/java/org/jnode/fs/xfs/XfsFileSystem.java

```diff
@@ -64,24 +85,36 @@ public final void read() throws FileSystemException {
         }
     }
 
+    public INode getINode(long iNodeNumber) {
+        ByteBuffer allocate = ByteBuffer.allocate(iNodeSize);
+
+        try {
+            this.getApi().read(iNodeSize * iNodeNumber, allocate);
+            return new INode(iNodeNumber, allocate.array() , 0);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading inode: " + iNodeNumber , e);
+        }
+    }
+
     @Override
     public long getTotalSpace() throws IOException {
-        return 0;
+        //TOTAL NUMBER OF BLOCKS..........
+        return superblock.getBlockSize() * superblock.getTotalBlocks();
     }
 
     @Override
     public long getFreeSpace() throws IOException {
-        return 0;
+        return superblock.getBlockSize() * superblock.getFreeBlocks();
     }
 
     @Override
     public long getUsableSpace() throws IOException {
-        return 0;
+        return superblock.getBlockSize() * (superblock.getTotalBlocks() - superblock.getFreeBlocks());
     }
 
     @Override
     public String getVolumeName() throws IOException {
-        return null;
+        return superblock.getName();
```

Comment:
`Superblock.getName()` will probably need to be updated to handle null terminated strings. `XfsFileSystemTypeTest` is printing a garbage value at the moment. It would be good to get the 0 values in the name be properly handled.

Reply:
change applied to avoid garbage, handling the null terminated strings.